include/merlin_hashtable.cuh

```diff
@@ -1070,51 +1113,54 @@ class HashTable {
    *
    * @return Number of keys loaded from file.
    */
-  size_type load(BaseKVFile<K, V, M, DIM>* file,
-                 const size_type buffer_size = 1048576,
-                 cudaStream_t stream = 0) {
-    std::unique_lock<std::shared_timed_mutex> lock(mutex_);
-    K* d_keys = nullptr;
-    V* d_vectors = nullptr;
-    M* d_metas = nullptr;
-    K* h_keys = nullptr;
-    V* h_vectors = nullptr;
-    M* h_metas = nullptr;
-    size_type pair_size = sizeof(K) + sizeof(M) + sizeof(V) * DIM;
-    size_type batch_pairs_num = (buffer_size + pair_size) / pair_size;
-
-    CUDA_CHECK(cudaMallocHost(&h_keys, sizeof(K) * batch_pairs_num));
-    CUDA_CHECK(cudaMallocHost(&h_vectors, sizeof(V) * batch_pairs_num * DIM));
-    CUDA_CHECK(cudaMallocHost(&h_metas, sizeof(M) * batch_pairs_num));
-    size_type nkeys = file->read(batch_pairs_num, h_keys, h_vectors, h_metas);
-    size_type counter = nkeys;
-    if (nkeys == 0) {
-      CUDA_FREE_POINTERS(stream, h_keys, h_vectors, h_metas);
-      CUDA_CHECK(cudaStreamSynchronize(stream));
+  size_type load(BaseKVFile<K, V, M, DIM>* file, cudaStream_t stream = 0) {
```

Comment:
@Lifann Change clarification: The buffer pool allows the hashtable to borrow an infinite *(until there is no more memory)* amount of buffers of size `buffer_size` *(it will never block requests, but may block on return if the `pending` queue is exhausted)*, there is no need to specify a temporary buffer size anymore for load/store operations. We just grab as many buffers as we need from the pool. `buffer_size` denotes the size of the batches. For you as external caller, there is no need to specify a `buffer_size` anymore.